File: app/src/main/java/lskycity/androidtools/BaseActivity.java
package lskycity.androidtools;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.NavigationView;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;

/**
 * Created by zhaofliu on 16-9-13.
 */
public class BaseActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener{

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        return false;
    }
}

File: app/src/main/java/com/lskycity/androidtools/app/BaseActivity.kt
package com.lskycity.androidtools.app

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * base activity for common thing.
 *
 * @author zhaofliu
 * @since 2016/2/13
 */
abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    override fun onSupportNavigateUp(): Boolean {
        if (!super.onSupportNavigateUp()) {
            supportFinishAfterTransition()
            return true
        }
        return false
    }
}
